### ISSUE TITLE
Short, gentle rant

### ISSUE THREAD

**Issue by gladiatr72** [saltstack/salt#31227]
```
TL;DR
```
I would love to see two things in the documentation:
 * some sort of indication when an auxiliary module is broken
 * whether or not said module works with the salt-master

I've been heavily into Saltstack for nearly two years now.  I think I contribute back quite a bit and put lots of plugs out for Salt where I can (Hey, if all you need is Fabric, use Fabric!)

On that note:

I have a gripe.

It might be unreasonable--hopefully you do not find it thus:

```
__context__
```

I just spent a week (7-days) putting together a set of  [Sentry] (http://getsentry.com) states.  It works perfectly.  It is integrated perfectly with the Rest of The Things (iptables rules, configures correctly with local or remote databases/caches, signed tls keys, etc)  I just got rid of the last (innocuously) failing state (Green is Good! :grinning: )

If I specify the right minion_id, I can kick off salt-cloud, walk away for 20 minutes and when I come back, I've got a perfectly working Sentry environment... that I have absolutely no use for.

I started working on this because of some issues I've run into with the 2015.8 releases, one of which caused the disk on the salt-master to fill up (logs) which caused various problems on our minion hosts.

"Sentry!  I've fiddled with that! And there's already Things that make Saltstack work with it!, WHEEE!"

----------
Today I find out the sad truth that the thing is pretty much broken for my purposes.  Maybe I didn't understand what `log_handlers` are supposed to be used for (not handling things on the salt-master??)
I started out with the latest `raven` client--found the posts wrt to needing to downgrade it for the time being.  

I wish I had time to Just Fix It.  

I understand that OSS projects are like little kids and old dogs--there always seems to be something wrong with them, right?  But the nature of a lot of these modules is that they require services that require  time/resource investment to set up--they've got their own proclivities--their own bugs, api-breaking changes, etc.

 Most of these (salt) module types are not nearly as a complex as the services they are designed to interface with, so a responsible, (non-bitchy, non-enterprise) user of saltstack is going to spend a bit of time making sure the complicated thing (elasticsearch, logstash, sentry, etc) is *also* configured properly before assuming that the salt module is not working correctly.

Anyway, you guys do a great job.  You are very forthcoming with letting folks know when a module has rotted or is non-functional when asked on the list or through an issue.

Unfortunately, this knowledge does not percolate down (up?) to the documentation and sometimes an issue just gets old, goes uncommented upon and then closed without an actual fix.

Blah.  Ok.  I'm done.  Thank you for reading :smile:

**Comment by jfindlay** [comment]
@gladiatr72, thanks for ranting.  Your first issue is obviously very difficult as it's nearly impossible and impractical for SaltStack to test each of the hundreds of integrations into other systems that have been written into the several plugin layers that salt has.  I've thought very long about this problem and am open to ideas.

The second issue seems to be much easier.  There are some plugin layers, like states that only run on minions and some, like runners, that only run on masters and some that run on both.  It should not be hard to be clear about which run where under which conditions.

**Comment by jfindlay** [comment]
@gladiatr72, also issues shouldn't be closed without adequate resolution regardless of the age.  Do you have any examples of this happening?

**Comment by gladiatr72** [comment]
Thanks for your reply, @jfindlay.  Not off the top of my head.  It may indeed be that the issues that I am remembering were blocked awaiting more data from the OP.  I'll give you a heads-up if I spot something like that (I sometimes peruse closed issues looking for info)

With regards to your previous response, I've got a pretty good handle (what I think of) as the outter-layer modules as to the master/minion functionality split.  Most of the time, at that layer, it is reasonably obvious as to whether or not a module/function is meant for minion-only or master-only use. 

runner/wheel/pillar/tops -> master
execution/util -> mix
states/grains -> minion

I think where the fog sets in, for me anyway, is in that middle-layer--returners and log handlers come to mind. I guess I assumed that the salt-master environment would be a first-tier customer for these kinds of modules--I find execution, grains and state modules to be (relatively) brain-dead simple to debug since they are running in a fairly low-concurrency environment; however, the salt-master, being extremely concurrent and touching All the Things is a bit tricker.  I'm hoping to have a block of time in the next couple months to revisit the log handler code--maybe bring it into line with both the minion and master environments (would sure be a useful thing to have!)

Anyway, thanks again.
